os/adding-swap.md

```diff
@@ -2,63 +2,61 @@
 
 Swap is the process of moving pages of memory to a designated part of the hard disk, freeing up space when needed. Swap can be used to alleviate problems with low-memory environments.
 
-By default Container Linux does not include a partition for swap, however one can configure their system to have swap.
+By default Container Linux does not include a partition for swap, however one can configure their system to have swap, either by including a dedicated partition for it or creating a swapfile.
 
 ## Managing swap with systemd
 
-A systemd-managed service can be used to create, enable, and disable a swapfile service on your system. Write, enable, and start the systemd unit file found in the next section to add swap to a Container Linux node.
+systemd provides a specialized `.swap` unit file type which may be used to activate swap. The below example shows how to add a swapfile and activate it using systemd.
 
-### Creating the systemd unit file
+### Creating a swapfile
 
-The following systemd unit creates and enables a service to manage a 1GiB swapfile. It should be written to `/etc/systemd/system/swap.service`.
+The following commands will make a 1GiB file suitable for use as swap.
 
+```sh
+mkdir -p /var/vm
+fallocate -l 1024m /var/vm/swapfile1
+mkswap /var/vm/swapfile1
 ```
+
+### Creating the systemd unit file
+
+The following systemd unit activates the swapfile we created. It should be written to `/etc/systemd/system/var-vm-swapfile1.swap`.
+
+```ini
 [Unit]
 Description=Turn on swap
 
-[Service]
-Type=oneshot
-Environment="SWAP_PATH=/var/vm" "SWAP_FILE=swapfile1"
-ExecStartPre=-/usr/bin/rm -rf ${SWAP_PATH}
-ExecStartPre=/usr/bin/mkdir -p ${SWAP_PATH}
-ExecStartPre=/usr/bin/touch ${SWAP_PATH}/${SWAP_FILE}
-ExecStartPre=/bin/bash -c "fallocate -l 1024m ${SWAP_PATH}/${SWAP_FILE}"
-ExecStartPre=/usr/bin/chmod 600 ${SWAP_PATH}/${SWAP_FILE}
-ExecStartPre=/usr/sbin/mkswap ${SWAP_PATH}/${SWAP_FILE}
-ExecStartPre=/usr/sbin/sysctl vm.swappiness=10
-ExecStart=/sbin/swapon ${SWAP_PATH}/${SWAP_FILE}
-ExecStop=/sbin/swapoff ${SWAP_PATH}/${SWAP_FILE}
-ExecStopPost=-/usr/bin/rm -rf ${SWAP_PATH}
-RemainAfterExit=true
+[Swap]
+What=/var/vm/swapfile1
 
 [Install]
 WantedBy=multi-user.target
 ```
 
-**Note** Ensure the block device containing your swapfile is configured to auto-mount on boot.
-
 ### Enable the unit and start using swap
 
-The following command enables and starts the new `swap` service.
+After creating the unit, it can be enabled. Optionally, the `swappiness` value can be modified if your system has sufficient memory for most tasks.
```

Comment:
delete 'Optionally' - Use `systemctl` to enable the unit once created. The `swappiness` value may be modified if the system has sufficient memory.